packages/iTwinUI-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -30,25 +30,54 @@ import { ComboBoxInput } from './ComboBoxInput';
 import { ComboBoxInputContainer } from './ComboBoxInputContainer';
 import { ComboBoxMenu } from './ComboBoxMenu';
 import { ComboBoxMenuItem } from './ComboBoxMenuItem';
+import { ComboBoxMultipleContainer } from './ComboBoxMultipleContainer';
 
-export type ComboBoxProps<T> = {
+const isMultipleEnabled = <T,>(
+  variable: (T | undefined) | (T[] | undefined),
+  multiple: boolean,
+): variable is T[] | undefined => {
+  return multiple;
+};
+
+// Type guard for multiple did not work
+const isSingleOnChange = <T,>(
+  onChange:
+    | (((value: T) => void) | undefined)
+    | (((value: T, event: SelectValueChangeEvent) => void) | undefined),
+  multiple: boolean,
+): onChange is ((value: T) => void) | undefined => {
+  return !multiple;
+};
+
+type ComboboxMultipleTypeProps<T> = {
   /**
-   * Array of options that populate the dropdown list.
+   * Enable multiple selection.
+   * @default false
    */
-  options: SelectOption<T>[];
+  multiple?: boolean;
   /**
    * Controlled value of ComboBox.
+   * If `multiple` is enabled, it is an array of values.
+   */
+  value?: T | T[];
+  /**
+   * Callback fired when selected value changes.
+   */
+  onChange?:
+    | ((value: T) => void)
+    | ((value: T, event: SelectValueChangeEvent) => void);
```

Comment:
so we don't have stricter conditional typing anymore? what problem were you running into?

Reply:
I was getting error in story that `type multiple: true is not compatible with false | undefined` (or something). I could add conditional typing back to try fix that issue some other way.